awx/main/models/unified_jobs.py

```diff
@@ -1512,3 +1514,47 @@ def log_lifecycle(self, state, blocked_by=None):
         else:
             msg = f"{self._meta.model_name}-{self.id} {state.replace('_', ' ')}"
         logger_job_lifecycle.debug(msg, extra=extra)
+
+#not doing the recursive logic right now , but just trying to make sure it returns something , the logic is basically what is used in the UI.
+    @property
+    def launched_by(self):
+        
+        #sdb.set_trace()
+        #checking if the current job was started by a workflow job.
+       
+        ancestor_job=self.ancestor_job
+
+        if hasattr(ancestor_job,'launch_type'):
+            if ancestor_job.launch_type=="manual":
+                return ancestor_job.created_by
+            elif ancestor_job.launch_type=="scheduled":
+                return ancestor_job.schedule
+            elif ancestor_job.launch_type=="dependency":
+                if ancestor_job.job_type_name=="project_update":
+                    return ancestor_job.project
+                elif ancestor_job.job_type_name=="inventory_update":
+                    return ancestor_job.inventory
+            else:
```

Comment:
For the webhook `launch_type`, we probably want to return `ancestor_job.unified_job_template`.

Reply:
I suspect,

- `'relaunch'` can only happen manually, so `.created_by`?
- `'sync'` is a project sync, so `.project`?
- `'scm'` is an inventory sync, so `.inventory`?
- `'callback'` says that it is started via a host callback, so `.limit`?  Except that that is a string so we're sad?

Reply:
Added `relaunch` condition.
Added 'webhook' condition. 
Added 'sync' condition. ( need to test it )
Added 'scm' condition . ( need to test it )